Mathlib/NumberTheory/Ostrowski.lean

```diff
@@ -55,4 +57,213 @@ lemma equiv_on_Nat_iff_equiv : (∃ c : ℝ, 0 < c ∧ (∀ n : ℕ , (f n)^c =
       (apply_nonneg f _), h x.den, ← MulRingNorm.apply_natAbs_eq,← MulRingNorm.apply_natAbs_eq,
       h (natAbs x.num)]
 
+/-!
+Throughout this file, `f` is an arbitrary absolute value.
+-/
+variable {f : MulRingNorm ℚ}
+
+open Rat.MulRingNorm
+
+section Non_archimedean
+
+-- ## Non-archimedean case
+
+/-- The mulRingNorm corresponding to the p-adic norm on ℚ. -/
+def mulRingNorm_padic (p : ℕ) [hp : Fact (Nat.Prime p)] : MulRingNorm ℚ :=
```

Comment:
It should not be necessary to name the instance argument (which can be written a bit shorter as `Fact p.Prime`, but this is a matter of taste).